plantdescriptionengine/src/main/java/eu/arrowhead/core/plantdescriptionengine/providedservices/pde_monitor/routehandlers/DtoUtils.java

```diff
@@ -0,0 +1,175 @@
+     *                    format.
+     * @return A list of connections adhering to the monitor package format.
+     */
+    private static List<ConnectionDto> mgmtToMonitor(final List<? extends Connection> connections) {
+        final List<ConnectionDto> result = new ArrayList<>();
+
+        for (final Connection connection : connections) {
+            final SystemPortDto consumerPort = new SystemPortDto.Builder()
+                .portName(connection.consumer().portName())
+                .systemId(connection.consumer().systemId())
+                .build();
+            final SystemPortDto producerPort = new SystemPortDto.Builder()
+                .portName(connection.producer().portName())
+                .systemId(connection.producer().systemId())
+                .build();
+
+            final ConnectionDto connectionCopy = new ConnectionDto.Builder()
+                .consumer(consumerPort)
+                .producer(producerPort)
+                .build();
+            result.add(connectionCopy);
+        }
+        return result;
+    }
+
+    /**
+     * Returns a Plant Description Entry System supplemented with monitor info.
+     *
+     * @param system      The source system which will be supplemented with
+     *                    monitor info.
+     * @param monitorInfo Object used for keeping track of inventory data of
+     *                    monitorable systems.
+     * @return A Plant Description Entry System with all the information
+     * contained in the {@code system} argument, supplemented with any relevant
+     * info in the {@code monitorInfo} argument.
+     */
+    private static SystemEntryDto extend(final PdeSystem system, final MonitorInfo monitorInfo) {
+
+        final List<MonitorInfo.Bundle> systemInfoList = monitorInfo.getSystemInfo(
+            system.systemName().orElse(null),
+            system.metadata()
+        );
+
+        final List<PortEntryDto> ports = new ArrayList<>();
+
+        // Add all ports. If there is monitor info bound to any of the ports,
+        // add it to the resulting port and remove it from the system info list.
+        for (final Port port : system.ports()) {
+
+            // 'consumer' defaults to false when no value is set:
+            final boolean isConsumer = port.consumer().orElse(false);
+
+            final PortEntryDto.Builder portBuilder = new PortEntryDto.Builder()
+                .portName(port.portName())
+                .serviceInterface(port.serviceInterface().orElse(null))
+                .serviceDefinition(port.serviceDefinition())
+                .consumer(isConsumer)
+                .metadata(port.metadata());
+
+            // Possibly add monitor info to ports where this system is the
+            // provider:
+            if (!isConsumer) {
+
+                for (final MonitorInfo.Bundle info : systemInfoList) {
+
+                    final boolean matchesServiceDefinition = info.serviceDefinition.equals(port.serviceDefinition());
```

Comment:
Similar to system name, serviceDefinitions is also normalized (toLowerCase().trim()) in the Service Registry. So, if any serviceDefinition here comes from the SR, that can cause issue in the comparison.

Reply:
Service definitions are now validated upon entering the PDE.